src/oop/user_defined_classes/medium/q2/MainTest.java

```diff
@@ -0,0 +1,111 @@
+package oop.user_defined_classes.medium.q2;
+
+import global.ObjectTest;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+import static global.tools.CustomAssertions._assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class MainTest {
+    // Java
+    private final String className = "Car";
+    private final String varColour = "colour";
+    private final String varPower = "horsePower";
+    private final String varYear = "yearMade";
+    private final String methodName = "improve";
+
+    private ObjectTest classInstance;
+
+    @BeforeEach
+    public void setUp() throws Throwable {
+        String classString = "oop.user_defined_classes.medium.q2." + className;
+        classInstance = new ObjectTest(classString);
+        String modifiedClassMessage =
+                "You have modified the provided portions of class " + className + ". Please revert them to the original state.";
+        Object[][] test = new Object[][]{
+                {"Red", 300, 2003},
+                {"Blue", 200, 2014},
+                {"Black", 100, 1986},
+                {"Yellow", 250, 2001}
+        };
+        for (int num = 0; num < test.length; num++) {
+            String colour = (String) test[num][0];
+            int horsePower = (int) test[num][1];
+            int yearMade = (int) test[num][1];
+            Object[][] arguments = {
+                    {colour, String.class},
+                    {horsePower, int.class},
+                    {yearMade, int.class}
+            };
+            Object instance = classInstance.createInstance(arguments);
+            _assertEquals(horsePower, classInstance.getFieldValue(instance, varPower), modifiedClassMessage);
+            _assertEquals(yearMade, classInstance.getFieldValue(instance, varYear), modifiedClassMessage);
+            _assertEquals(colour, classInstance.getFieldValue(instance, varColour), modifiedClassMessage);
+        }
+    }
+    @Test
+    public void classHasRequiredFields() {
+        assertTrue(classInstance.hasField(varColour),
+                "Your " + className + " class is missing the " + varColour + " attribute, or it is spelled incorrectly.");
+        assertTrue(classInstance.hasField(varColour, String.class),
```

Comment:
You are not checking that the varColour attribute exists

Reply:
It is checking if the field exists I don't get the issue